pkgs/data/fonts/nerd-fonts/update.js

```diff
@@ -0,0 +1,74 @@
+const RELEASE_VERSION_FILENAME = "release-version.json";
+const RELEASE_INFO_URL = "https://api.github.com/repos/ryanoasis/nerd-fonts/releases/latest";
+const PKGS_INFO_FILENAME = "fonts.json";
+
+const getFontsInfoUrl = (tag) => `https://raw.githubusercontent.com/ryanoasis/nerd-fonts/refs/tags/${tag}/bin/scripts/lib/fonts.json`;
+
+const fetchJSON = (url) => JSON.parse(std.urlGet(url).split("\r\n").at(-1));
+
+const convertAttrName = (formerAttrName) => {
+    if (/^\d/u.test(formerAttrName)) {
+        return "_" + formerAttrName;
+    } else {
+        return formerAttrName;
+    }
+};
+
+const convertVersion = (formerVersion, dateString) => {
+    if (/^\d/u.test(formerVersion)) {
+        return formerVersion;
+    } else {
+        return "0-unstable-" + dateString;
+    }
+};
+
+let formerReleaseVersion = JSON.parse(std.loadFile(RELEASE_VERSION_FILENAME));
+let releaseInfo = fetchJSON(RELEASE_INFO_URL);
+let tag = releaseInfo.tag_name;
+let releaseVersion = tag.split("v")[1];
+
+if (releaseVersion === formerReleaseVersion) {
+    std.exit(1);
+} else {
+    let fonts = fetchJSON(getFontsInfoUrl(tag)).fonts;
+
+    let pkgsInfo = fonts.map(font => {
+        let pname = font.caskName.toLowerCase();
+
+        let url = releaseInfo.assets
+            .find(asset => asset.name === font.folderName + ".tar.xz")
+            .browser_download_url;
+
+        let sha256 = (() => {
+            let f = std.popen(`nix-prefetch-url ${url} --unpack`, "r");
+            let result = f.getline();
+            f.close();
+            return result;
+        })();
+
+        return ({
+            pname: pname,
+            attrName: convertAttrName(pname),
+            dirName: font.patchedName.replaceAll(/\s+/gu, ""),
+            originalName: font.unpatchedName,
+            declaresRFN: font.RFN,
```

Comment:
Hmm this is a boolean, and on a 2nd thought I don't think there is much value in storing these values in Nixpkgs. Now I also see that not all fonts have an RFN according to [upstream's online list](https://www.nerdfonts.com/font-downloads). Not only that, perhaps what I thought earlier that was the RFN, is actually the "font family". Please excuse me for not being clear earlier, but I'd suggest:

Make the `fonts.json` have the following structure:

```
<font-family> = {
  ...
}
```

And these could actually become the `nerd-fonts.<font-family>` attributes, so you won't need `convertAttrName`. I also wonder if there is any value in the `url`? The logic of inferring it from another attribute is pretty trivial and perhaps would be easier to maintain if will be part of `default.nix`.

In general, I'd like to minimize the size of such generated files, and maintain most logic in Nix.

Reply:
According to the [Guideline](https://github.com/NixOS/nixpkgs/tree/master/pkgs#package-naming), it is expected that the `pname`

> _should_ be identical to the upstream package name.

and the attribute name of a new package

> _should_ be the same as the value in `pname`.

And according to the upstream [script for generating Homebrew casks](https://github.com/ryanoasis/nerd-fonts/blob/master/bin/scripts/generate-casks.sh), the package name was determined with
```sh
caskbasename=$(jq -r ".fonts[] | select(.folderName == \"${basename}\") | .caskName" "${scripts_root_dir}/lib/fonts.json" | head -n 1)
#...
caskname="font-${caskbasename}-nerd-font"
```
Moreover, there is such copyright issue I referred [above](https://github.com/NixOS/nixpkgs/pull/354543#discussion_r1835480927).

So IMO it would be best to just use `caskName` for `pname` as well as the attribute name.

On the other hand, getting the family name of the fonts is not so trivial a task. According to the same upstream script ([lines](https://github.com/ryanoasis/nerd-fonts/blob/master/bin/scripts/generate-casks.sh#L78)), it would need `fc-query` as well as _the font files_ available.

Reply:
It is the logic of Nix that each hash is corresponding to a fixed-output derivation, and thus a URL (not so strictly, though, since we use `fetchzip`). Pairing the hash and the URL together in the "manifest" ensures and emphasize the connection.

Also, from the perspective of programming, the URL is found by searching against the filename, and the hash is computed from the file fetched with that URL. The cleanest logic I can imagine is to just put them together in the same JSON output.

Reply:
The font families have been added!